### PR TITLE
Direct input type

### PR DESCRIPTION
**Pull request by Priyanka-ashok**
#26

### CODE REVIEW

File: client/src/migrator.js [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+const selectOptions = [{ label: "Story", value: "Story" }];
+
+const validateOptions = [
+  { label: "Direct text input", value: "Direct text input" }
+];
+export class Migrator extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      validatetype: null,
```
Comment (Robert-Christopher) [inline client/src/migrator.js]: Camel here ?

File: client/src/migrator.js [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+    e.preventDefault();
+    this.setState({
+      disabled: !this.state.disabled,
+      isLoading: true
+    });
+    const data = {
+      type: this.state.selecttype.value,
+      data: this.state.text
+    };
+    fetch("http://localhost:3000/api/validate", {
```
Comment (Robert-Christopher) [inline client/src/migrator.js]: localhost ?

File: server/package.json [hunk 3]
```diff
@@ -31,6 +31,7 @@
     "version": "standard-version",
     "reset": "git clean -dfx && git reset --hard && npm i",
     "clean": "trash build test",
+    "jest": "jest",
```
Comment (Robert-Christopher) [inline server/package.json]: Why this is defined separately ? Can we not reuse `test` above ?

File: server/src/lib/handlers/validator.spec.ts [hunk 4]
```diff
@@ -0,0 +1,264 @@
+import { readFileSync } from 'fs';
+import { join } from 'path';
+import { generateJsonSchema, validateJson, validator } from './validator';
+
+const testSchema =  JSON.parse(readFileSync(join(__dirname,'..','..','..','test-data','test_schema.json'), 'utf8'));
+const authorSchema = generateJsonSchema(join(__dirname, '..','..','..','test-data','editor-test.ts'), 'AuthorTest');
+const sectionSchema = generateJsonSchema(join(__dirname, '..','..','..','test-data','editor-test.ts'), 'SectionTest');
+const storySchema = generateJsonSchema(join(__dirname, '..','..','..','test-data','editor-test.ts'), 'StoryTest');
```
Comment (Robert-Christopher) [inline server/src/lib/handlers/validator.spec.ts]: Code repeated ? Please refactor ?

File: server/src/lib/server.ts [hunk 5]
```diff
@@ -1,61 +1,86 @@
...
+  'build',
+  'main',
+  'lib',
+  'editor-types.d.ts'
+);
 
 export const app = express();
 app.use(compression());
-app.use(bodyParser.json({ limit: "1mb" }));
+app.use(bodyParser.json({ limit: '1mb' }));
```
Comment (Robert-Christopher) [inline server/src/lib/server.ts]: Should we have this in a config ?

File: server/src/lib/server.ts [hunk 6]
```diff
@@ -1,61 +1,86 @@
...
       callback(null, true);
       return;
     }
-    if (origin.endsWith("quintype.com") ||
-       (process.env.NODE_ENV !== 'production' && origin.startsWith("http://localhost:"))) {
-      callback(null, true)
+    if (
+      origin.endsWith('quintype.com') ||
+      (process.env.NODE_ENV !== 'production' &&
+        origin.startsWith('http://localhost:'))
```
Comment (Robert-Christopher) [inline server/src/lib/server.ts]: localhost ?

File: server/src/lib/server.ts [hunk 7]
```diff
@@ -1,61 +1,86 @@
...
-    if(!origin) {
+    if (!origin) {
       callback(null, true);
       return;
     }
-    if (origin.endsWith("quintype.com") ||
-       (process.env.NODE_ENV !== 'production' && origin.startsWith("http://localhost:"))) {
-      callback(null, true)
+    if (
+      origin.endsWith('quintype.com') ||
```
Comment (Robert-Christopher) [inline server/src/lib/server.ts]: Should this be in a config ?

File: server/test-data/editor-test.ts [hunk 8]
```diff
@@ -0,0 +1,21 @@
...
+
+export interface StoryTest {
+  /** Name of story */
+  readonly name: string;
+  /** List of sections story belongs to */
+  readonly sections: ReadonlyArray<SectionTest>;
+
+  /** List of story authors */
+  readonly authors: ReadonlyArray<AuthorTest>;
+}
```
Comment (Robert-Christopher) [inline server/test-data/editor-test.ts]: I think this is already defined here https://github.com/quintype/quintype-node-migration-helpers/blob/master/src/lib/editor-types.ts#L124

File: client/src/migrator.js (hunk 1 above)
Comment (Priyanka-ashok) [reply client/src/migrator.js]: Done

File: server/src/lib/server.ts (hunk 7 above)
Comment (Priyanka-ashok) [reply server/src/lib/server.ts]: As discussed, we will fix it later

File: server/src/lib/server.ts (hunk 5 above)
Comment (Priyanka-ashok) [reply server/src/lib/server.ts]: As discussed, we will fix it later

File: client/src/migrator.js (hunk 2 above)
Comment (Priyanka-ashok) [reply client/src/migrator.js]: Removed

File: server/src/lib/handlers/validator.spec.ts (hunk 4 above)
Comment (Priyanka-ashok) [reply server/src/lib/handlers/validator.spec.ts]: Refactored

File: server/test-data/editor-test.ts (hunk 8 above)
Comment (Priyanka-ashok) [reply server/test-data/editor-test.ts]: No it is defined in different folder (From client-migration)

File: server/package.json (hunk 3 above)
Comment (Priyanka-ashok) [reply server/package.json]: Merged into test , Need to run "npm run test" only

File: server/src/lib/fetch-lighthouse.ts [hunk 9]
```diff
@@ -21,7 +21,6 @@ export async function fetchLighthouse(url: string): Promise<LighthouseResults> {
   }
 }
```
Comment (Robert-Christopher) [inline server/src/lib/fetch-lighthouse.ts]: Can we get this out of this PR, we should be only working on migration validator for now

File: server/src/lib/handlers/validate-url-handler.ts [hunk 10]
```diff
@@ -29,7 +29,6 @@ export async function validateUrlHandler(req: Request, res: Response): Promise<v
       links
     });
   } catch (error) {
-    // tslint:disable-next-line: no-console
     console.error(error.stack || error);
```
Comment (Robert-Christopher) [inline server/src/lib/handlers/validate-url-handler.ts]: Refer previous comment on migration validator changes only

File: server/src/lib/handlers/validator.spec.ts [hunk 11]
```diff
@@ -0,0 +1,273 @@
+import { readFileSync } from 'fs';
```
Comment (Robert-Christopher) [inline server/src/lib/handlers/validator.spec.ts]: Complete file needs refactoring

**Pull request merged**